packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx

```diff
@@ -23,7 +23,7 @@ import styles from "./RichTextEditor.module.scss"
 
 export interface BaseRichTextEditorProps
   extends OverrideClassName<Omit<HTMLAttributes<HTMLDivElement>, "onChange">> {
-  onChange: (content: EditorContentArray) => void
+  onChange: (content: ProseMirrorState.EditorState) => void
```

Comment:
I am hesitant to introduce this breaking change. I'd just provide an additional data leaving 'content' alone as we have a number of other systems integrating RTE, which means they will have to update them too when they upgrade.

Reply:
Cool thanks. Just saw that thread now.

@dougmacknz @gyfchong 
Some further comments on MD vs Prosemirror, not particularly relating to this PR:

One thing that popped up during the solution preview design meeting was that Performance products rely on MD quite a lot due to how it's supported out of the box in Slack etc. When we send slack message with the RTE content, it's rendered correctly in most part according to the MD format.

We also had opinions saying that MD is the more accepted schema out in the wild than prose-mirror.

Reply:
I think this needs further thought as to the future of MD, something we can't resolve here. But my overall thought is to answer the question "what does CultureAmp need?" rather than follow a more accepted schema because it's more accepted. For example, If our use case for MD stops at Slack and the rest of the platform needs ProseMirror features, then we can concentrate our adapters in that specific communication area, converting ProseMirror in a very isolated circumstance.

Regarding the breaking change, we should add migration notes for those upgrading on how to achieve the same outcome as they had before this change.

Reply:
@gyfchong 
Sure we can have more discussion about MD. Not in the scope for this PR.